src/include/dns-server/dialog-main.rb

```diff
@@ -1898,12 +1898,36 @@ def SaveAndRestart(event)
 
     # Writes DNS server settings and saves the service
     #
-    # @note currently, the DnsServer is a Perl module, reason why the write of
-    # settings is being performed in two steps.
+    # As side effect, it also sets/resets the local forwarder
     #
     # @return [Boolean] true if settings are saved successfully; false otherwise
     def write_settings
-      DnsServer.Write && service.save
+      return false unless DnsServer.Write
+
+      performed_action = service.action
+      service_saved = service.save(keep_state: Mode.auto)
+
+      reset_local_forwarder(performed_action, service_saved)
+
+      service_saved
+    end
+
+    # Resets the local forwarder to "resolver"
+    #
+    # Local forwarder will be changed to "resolver" if
+    #
+    #   * its current value is not "bind"
+    #   * service was stopped
+    #   * service cannot be started, restarted or reloaded
+    #
+    # @param performed_action [Symbol] action executed
+    # @param service_saved [Boolean] result of peforms the service#save
```

Comment:
typo: peforms -> performs. Anyway, if you want to keep this parameter, I would change the description for something like "whether the service was successfully saved or not".